### PR TITLE
Cross scala 3

### PR DESCRIPTION
**Pull request by fedefernandez**
# Overview

This PR brings cross-compilation with Scala 3 to memeid. The trickiest part was the memeid4s-literal, where a new Scala 3 macro was implemented.

I've also added the `sbt-projectmatrix` plugin and re-configured the modules. The main purpose is to be able to maintain different modules with different `crossScalaVersions`.

## Caveats

Some caveats from the current upgrade:

1. The build fails due to a misconfiguration with scalafix. For some reason, the sbt-scalafix-defaults plugin is bringing Scala 2 config in a Scala 3 build. Locally works fine. For now, the decision I made is to remove the plugin. The error can be seen [here](https://github.com/47degrees/memeid/actions/runs/4079039438/jobs/7029976462)
2. The mdoc compilation fails with doobie code and Scala 3. In this PR we're just compiling the documentation against Scala 2.12 and Scala 2.13. I'll create an issue with the error once this is merged. 

```auto
error:
Exception occurred while executing macro expansion.
java.lang.NullPointerException
	at doobie.util.PosPlatform$.originImpl(PosPlatform.scala:21)

info: Compiled in 20.61s (1 error)
[error] java.lang.RuntimeException: mdoc failed
[error] 	at scala.sys.package$.error(package.scala:27)
[error] 	at mdoc.SbtMain$.main(Main.scala:30)
[error] 	at mdoc.SbtMain.main(Main.scala)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error] stack trace is suppressed; run last microsite3 / Compile / runMain for the full output
[error] (microsite3 / Compile / runMain) mdoc failed
[error] Total time: 35 s, completed Feb 2, 2023, 11:03:25 PM
```

### CODE REVIEW

**Comment by fedefernandez** [comment]
The doobie `mdoc` is still to be implemented

**Comment by fedefernandez** [comment]
Pending to improve the build.sbt

**Comment by fedefernandez** [comment]
@gutiory this is ready for review. Thanks

File: project/plugins.sbt [hunk 1]
```diff
@@ -10,7 +11,6 @@ addSbtPlugin("com.alejandrohdezma"               % "sbt-github-mdoc"          %
 addSbtPlugin("com.alejandrohdezma"               % "sbt-mdoc-toc"             % "0.4.0")
 addSbtPlugin("com.alejandrohdezma"               % "sbt-modules"              % "0.2.0")
```
Comment (gutiory) [inline project/plugins.sbt:12]: I think you can remove `sbt-modules` plugin.
```suggestion
```

**Review by gutiory — APPROVED** [review]
Astonishing work @fedefernandez!. I've just left a minor comment.

**Comment by fedefernandez** [comment]
@gutiory I've applied your changes. I'm not sure why the patch is complaining, but the changes look good to me. Could you please review it again?

**Review by gutiory — APPROVED** [review]

**Pull request merged**